scripts/check-precommit-reqs

```diff
@@ -11,23 +11,27 @@
 ##==============================================================================
 
 ## Check whether source code has been formatted (diff outputs '@@'):
-if ./scripts/format-code --whatif | grep -q "@@"; then
+if [[ $# -ne 0 ]]; then
+    files="$*"
+fi
+
+if ./scripts/format-code --whatif $([[ "$files" ]] && echo "--files=${files// /,}") | grep -q "@@"; then
     echo ""
     echo "Commit failed: please run ./scripts/format-code"
     echo ""
     exit 1
 fi
 
 # Check whether all sources have the license header:
-if ! ./scripts/check-license; then
+if ! ./scripts/check-license "$files"; then
     echo ""
     echo "Commit failed: please add license headers to the above files"
     echo ""
     exit 1
 fi
 
 # Check whether all scripts pass ShellCheck:
-if ! ./scripts/check-linters; then
+if ! ./scripts/check-linters "$files"; then
```

Comment:
Same issue here. When I gave an explicit file `format-code`, it did check it, and errored (as these change don't pass shellcheck):

```
In ./scripts/format-code line 238:
    echo $list | tr ' ' '\n' > $tempfile
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                               ^-- SC2086: Double quote to prevent globbing and word splitting.


In ./scripts/format-code line 240:
    for f in ${files[@]}; do
             ^-- SC2068: Double quote array expansions to avoid re-splitting elements.


In ./scripts/format-code line 243:
            file_list=$(echo $file_list $new_file)
                      ^-- SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.
                                        ^-- SC2086: Double quote to prevent globbing and word splitting.


Commit failed: please run ./scripts/check-linters
```